### PR TITLE
Substituting Fetch with XHR

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: src/repository/summary.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
-    );
+const getPromise = new Promise(function(resolve, reject) {
+    const request = new XMLHttpRequest();
+
+    request.onload = function() {
+        resolve(JSON.parse(this.responseText));
+    }
+    
+    request.onerror = function() {
+        reject();
```
Comment (mike-kaufman) [inline src/repository/summary.js]: do yo uwant to pass the error as the rejection reason.  i.e.., `reject(err)`

**Review by mike-kaufman — APPROVED** [review]
minor comment.

File: src/repository/summary.js (hunk 1 above)
Comment (davidkpiano) [reply src/repository/summary.js]: Sure

**Review by mike-kaufman — APPROVED** [review]

**Review by philliphoff — APPROVED** [review]
LGTM

**Comment by legomushroom** [comment]
LGTM

**Review by legomushroom — APPROVED** [review]
LGTM

**Comment by avanderhoorn** [comment]
Not that it really matters but interested in why the change?

**Comment by davidkpiano** [comment]
> Not that it really matters but interested in why the change?

Regression - Older versions of Safari (e.g., the one currently installed on my MBP) don't support `fetch` and we can't expect developers to have the absolute latest version of Safari. Especially on my 2015 MBP, it's not trivial to update Safari. Because `fetch` didn't exist, the HUD completely failed to render.

**Pull request merged**